src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -401,23 +401,38 @@ void AppHost::LastTabClosed(const winrt::Windows::Foundation::IInspectable& /*se
         _HideNotificationIconRequested();
     }
 
+    // We don't want to try to save layouts if we are about to close
+    _getWindowLayoutThrottler.reset();
+    _windowManager.GetWindowLayoutRequested(_GetWindowLayoutRequestedToken);

```

Comment:
The issue is that the event handler can get called _during_ the destructor for the `AppHost` so technically `this` is still valid, but half of the state is gone. Removing the event handler makes it so that it can't get called at all.